app/models/user.rb

```diff
@@ -0,0 +1,68 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id                     :integer          not null, primary key
+#  email                  :string           default(""), not null
+#  encrypted_password     :string           default(""), not null
+#  reset_password_token   :string
+#  reset_password_sent_at :datetime
+#  remember_created_at    :datetime
+#  sign_in_count          :integer          default(0), not null
+#  current_sign_in_at     :datetime
+#  last_sign_in_at        :datetime
+#  current_sign_in_ip     :inet
+#  last_sign_in_ip        :inet
+#  confirmation_token     :string
+#  confirmed_at           :datetime
+#  confirmation_sent_at   :datetime
+#  unconfirmed_email      :string
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  name                   :text             not null
+#  msf_location_id        :integer
+#  external_location      :text
+#  role                   :enum             default("normal_user"), not null
+#
+# Indexes
+#
+#  index_users_on_confirmation_token    (confirmation_token) UNIQUE
+#  index_users_on_email                 (email) UNIQUE
+#  index_users_on_msf_location_id       (msf_location_id)
+#  index_users_on_reset_password_token  (reset_password_token) UNIQUE
+#
+
+class User < ActiveRecord::Base
+  # Include devise modules. Others available are:
+  # :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable, :recoverable,
+         :rememberable, :trackable, :validatable, :confirmable
+
+  enum role: {
```

Comment:
It does limit them to only being one thing, but I've assumed that this reflects MSF's structure. I don't think anyone except dedicated users like Kim will be admins. With regards to being a contributor, I'm assuming that these will be a list managed on the study itself, so you can have a RM/PI role but also be in a different study's list of contributors and thus allowed to do what contributor's do there. 

I guess since I've added a PI/RM field on studies, we could actually infer all of the powers from that, rather than needing the roles at all (except for an "is_admin" flag/role), what do you think? I guess this will allow us to find all of the PIs/RMs more easily for stuff like the yearly mail-outs...